.github/workflows/checkmake.yaml

```diff
@@ -0,0 +1,26 @@
+name: CheckMake
+on:
+  pull_request:
+    branches:
+      - master
+      - release-*
+
+# cancel the in-progress workflow when PR is refreshed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  checkmake:
+    name: CheckMake
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Run CheckMake
+        uses: Uno-Takashi/checkmake-action@v2
+        with:
+          Makefile: Makefile
```

Comment:
Does this only run on the makefile in the root folder? Can we also run it against the makefiles in other folders?

Reply:
@travisn wrote:
> Does this only run on the makefile in the root folder?
yes currently only against the main Makefile.

> Can we also run it against the makefiles in other folders?

We can, but locally, these files show errors:
```console
build/release/Makefile images/Makefile images/ceph/Makefile
```

Reply:
Ok as discussed offline, let's go ahead with just the main Makefile for now.